src/app/api/services/dot-contentlet/dot-contentlet.service.spec.ts

```diff
@@ -0,0 +1,91 @@
+import { DOTTestBed } from '../../../test/dot-test-bed';
+import { DotContentletService } from './dot-contentlet.service';
+import { ConnectionBackend, ResponseOptions, Response } from '@angular/http';
+import { MockBackend } from '@angular/http/testing';
+import { mockDotContentlet } from '../../../test/dot-contentlet.mock';
+import { StructureTypeView } from '../../../shared/models/contentlet/structure-type-view.model';
+
+let lastConnection: any;
+const MAIN_CONTENT_TYPES = ['CONTENT', 'WIDGET', 'FORM', 'FILEASSET', 'HTMLPAGE'];
+
+function mockConnectionContentletResponse(): void {
+    return lastConnection.mockRespond(
+        new Response(
+            new ResponseOptions({
+                body: {
+                    entity: mockDotContentlet
+                }
+            })
+        )
+    );
+}
+
+function isRecentContentType(type: StructureTypeView): boolean {
+    return type.name.startsWith('RECENT');
+}
+
+function isMainContentType(type: StructureTypeView): boolean {
+    return MAIN_CONTENT_TYPES.indexOf(type.name) > -1;
+}
+
+function isMoreContentType(type: StructureTypeView): boolean {
+    return !isRecentContentType(type) && !isMainContentType(type);
+}
+
+describe('DotContentletService', () => {
+    beforeEach(() => {
+        this.injector = DOTTestBed.resolveAndCreate([DotContentletService]);
+        this.dotContentletService = this.injector.get(DotContentletService);
+        this.backend = this.injector.get(ConnectionBackend) as MockBackend;
+        this.backend.connections.subscribe((connection: any) => (lastConnection = connection));
+    });
+
+    it('should call the BE with correct endpoint url and method for getContentTypes()', () => {
+        this.dotContentletService.getContentTypes().subscribe((structures: StructureTypeView[]) => {
+            expect(structures).toEqual(mockDotContentlet);
+        });
+        mockConnectionContentletResponse();
+        expect(lastConnection.request.method).toBe(0); // 0 is GET method
+        expect(lastConnection.request.url).toContain(`v1/contenttype/basetypes`);
+    });
+
+    it('should get the Main Content Types for getMainContentTypes()', () => {
+        this.dotContentletService.getMainContentTypes().subscribe((structures: StructureTypeView[]) => {
+            const types = mockDotContentlet.filter((structure: StructureTypeView) => isMainContentType(structure));
```

Comment:
We don't need so this here, we already know witch contents are main without call the 'isMainContentType' function